### DISCUSSION TITLE
How to change the radius of the standard.Circle?

### DISCUSSION THREAD

**Discussion by xxxHunterxxx19** [clientIO/joint#2666] [Q&A]
### Introduction

Through rezise, halo, freetransform, the size of the circle does not change. What to do? How do I change the radius manually or how do I configure the halo to change the size of the circle?

### Does your question relate to JointJS or JointJS+. Select both if applicable.

JointJS, JointJS+

**Comment by kumilingus** [comment]
How did you define the circle element?

You need to know how to define the radius of the circle to be relative to the size of the model. The `ui.Halo` and `ui.FreeTransform` changes the model size and position.

You can use `calc()` expressions to do this: https://resources.jointjs.com/tutorial/special-attributes

Here's an example:
```ts
const MyCircle = dia.Element.define('MyCircle', {
    size: {
        width: 100,
        height: 100
    },
    attrs: {
        body: {
            cx: 'calc(s/2)',
            cy: 'calc(s/2)',
            r: 'calc(s/2)',
            stroke: '#333333',
            fill: '#FFFFFF'
        },
    }
}, {
    markup: `
       <circle @selector="body"/>       
    `
});
```

*Note: if you want to resize an element into an ellipse (different `rx` and `ry`), you must use `<ellipse/>` instead.*
